### DISCUSSION TITLE
feature question "Type-safe static CSS"

### DISCUSSION THREAD

**Discussion by roccho-dev** [mayank99/ecsstatic#19] [Q&A] [answered]
ecsstatic generates css file, like vanilla-extraction?

https://vanilla-extract.style/

```
Type-safe static CSS
All styles generated at build time — just like [Sass](https://sass-lang.com/), [LESS](https://lesscss.org/), etc, but with the power of TypeScript.
```

OR
this feature means above as same?

https://github.com/mayank99/ecsstatic#atomic-classes

**Comment by mayank99** [comment, marked as answer]
Yes, all styles are statically extracted into a CSS file at build time. It's mentioned in the first line in readme.
